src/Foundation/NSObject2.cs

```diff
@@ -231,16 +231,14 @@ public void Dispose ()
 			GC.SuppressFinalize (this);
 		}
 
-		static T AllocateNSObject<T> (IntPtr handle) where T : NSObject
-		{
-			var obj = (T) RuntimeHelpers.GetUninitializedObject (typeof (T));
-			obj.handle = handle;
-			obj.flags = Flags.NativeRef;
-			return obj;
-		}
-
 		internal static IntPtr CreateNSObject (IntPtr type_gchandle, IntPtr handle, Flags flags)
 		{
+#if NET
+			if (Runtime.IsManagedStaticRegistrar) {
+				throw new System.Diagnostics.UnreachableException ();
+			}
```

Comment:
Ideally we should figure out how to make the linker trim CreateNSObject away, but I like this solution in the mean time!